aiidalab_launch/instance.py

```diff
@@ -286,34 +286,45 @@ async def echo_logs(self) -> None:
     @staticmethod
     async def _init_scripts_finished(container: Container) -> None:
         loop = asyncio.get_event_loop()
-        LOGGER.debug("Waiting for init services to finish...")
+        LOGGER.info("Waiting for init services to finish...")
         result = await loop.run_in_executor(
             None, container.exec_run, "wait-for-services"
         )
         if result.exit_code != 0:
             raise FailedToWaitForServices(
                 "Failed to check for init processes to complete."
             )
-        LOGGER.debug("Init services finished.")
+        LOGGER.info("Init services finished.")
 
     @staticmethod
     async def _notebook_service_online(container: Container) -> None:
         loop = asyncio.get_event_loop()
-        LOGGER.debug("Waiting for notebook service to become reachable...")
+        LOGGER.info("Waiting for notebook service to become reachable...")
+        proto = "http"
         while True:
             try:
                 LOGGER.debug("Curl notebook...")
                 result = await loop.run_in_executor(
                     None,
                     container.exec_run,
-                    "curl --fail-early --fail --silent --max-time 1.0 http://localhost:8888",
+                    f"curl --fail-early --fail --silent --max-time 1.0 {proto}://localhost:8888",
                 )
                 if result.exit_code == 0:
-                    LOGGER.debug("Notebook service reachable.")
+                    LOGGER.info("Notebook service reachable.")
                     return  # jupyter is online
                 elif result.exit_code in (7, 28):
                     await asyncio.sleep(2)  # jupyter not yet reachable
                     continue
+                elif result.exit_code == 56 and proto == "http":
+                    proto = "https"
+                    LOGGER.info("Trying to connect via HTTPS.")
+                    continue
```

Comment:
According to the curl manual, exit code 56 indicates a "[f]ailure in receiving network data." To my understanding this _can_ be caused by the wrong protocol, but there could of course also be other reasons for this failure mode. How confident are you that switching the protocol at this point is _always_ the right move?

Should we maybe try both protocols?

Reply:
I am not sure what you're proposing. 

> How confident are you that switching the protocol at this point is always the right move?
Not confident at all. I was hoping that curl will give me a more specific error when trying the wrong protocol, but alas this is what I am getting in Ubuntu 20.04 and curl 7.68. Note however that this PR does not change behavior with respect to the http protocol. If we get this error code, we simply try https once and then bail out if that doesn't work (so in the worst case we just added an extra useless ping).

Reply:
Yes, I misread the logic earlier and thought the protocol was essentially permanently switched. I think this is good like this. 👍

Reply:
Great, thanks for the review! I've addressed the other bits.